test/flows/maternity_paternity_pay_leave_flow_test.rb

```diff
@@ -0,0 +1,552 @@
+require "test_helper"
+require "support/flow_test_helper"
+
+class MaternityPaternityPayLeaveFlowTest < ActiveSupport::TestCase
+  include FlowTestHelper
+
+  setup do
+    testing_flow MaternityPaternityPayLeaveFlow
+  end
+
+  should "render a start page" do
+    assert_rendered_start_page
+  end
+
+  context "question: two_carers" do
+    setup { testing_node :two_carers }
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node of due_date for any response" do
+        assert_next_node :due_date, for_response: "yes"
+      end
+    end
+  end
+
+  context "question: due_date" do
+    setup do
+      testing_node :due_date
+      add_responses two_carers: "yes"
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node of employment_status_of_mother for any response" do
+        assert_next_node :employment_status_of_mother, for_response: "2016-1-1"
+      end
+    end
+  end
+
+  context "question: employment_status_of_mother" do
+    setup do
+      testing_node :employment_status_of_mother
+      add_responses two_carers: "yes",
+                    due_date: "2016-1-1"
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node of employment_status_of_partner for any response and there are two carers" do
+        assert_next_node :employment_status_of_partner, for_response: "employee"
+      end
+
+      %w[employee worker].each do |response|
+        should "have a next node of mother_started_working_before_continuity_start_date for a '#{response}' response and there is a single carer" do
+          add_responses two_carers: "no"
+          assert_next_node :mother_started_working_before_continuity_start_date, for_response: response
+        end
+      end
+
+      %w[self-employed unemployed].each do |response|
+        should "have a next node of mother_worked_at_least_26_weeks for a '#{response}' response and there is a single carer" do
+          add_responses two_carers: "no"
+          assert_next_node :mother_worked_at_least_26_weeks, for_response: response
+        end
+      end
+    end
+  end
+
+  context "question: employment_status_of_partner" do
+    setup do
+      testing_node :employment_status_of_partner
+      add_responses two_carers: "yes",
+                    due_date: "2016-1-1",
+                    employment_status_of_mother: "employee"
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node of mother_started_working_before_continuity_start_date for any response" do
```

Comment:
So here you probably do want a test for each employment status of the mother, as it does affect the flow.
```suggestion
      should "have a next node of mother_started_working_before_continuity_start_date for any response when the mother is an employee" do
```